applications/GeoMechanicsApplication/tests/test_dirichlet_release.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import test_helper
+
+
+class KratosGeoMechanicsDirichletReleaseTests(KratosUnittest.TestCase):
+    """
+    This class contains a test for displacement Dirichlet boundary condition which is released in a second stage.
+    """
+
+    def test_dirichlet_release(self):
+        """
+        Single element test in 2 stages. The incremental elastic material should show
+        linear increase in stress_yy in the first step. In the second stage the Dirichlet condition
+        is removed, meaning the stress_yy should be 0.
+        H = 1 [m]
+        Young's modulus E = 10000 [N/m^2]
+        stage 1) elongate 0.1 [m]
+        stage 2) release the prescribed displacement, meaning the stress should be 0
+        """
+        test_name = "dirichlet_release"
+        project_path = test_helper.get_file_path(test_name)
+        n_stages = 2
+        stages = test_helper.get_stages(project_path, n_stages)
+
+        # name of output file
+        output_file_names = [
+            os.path.join(project_path, f"dirichlet_release_stage{i + 1}.post.res")
+            for i in range(n_stages)
+        ]
+        output_data = []
+
+        # run stages and get results
+        for stage, output_file_name in zip(stages, output_file_names):
+            stage.Run()
+            reader = test_helper.GiDOutputFileReader()
+            output_data.append(reader.read_output_from(output_file_name))
+
+        # Expected stress, resulting from the compression, Poisson effects (nu = 0.2) and
+        # the strain in the x-direction (partially induced by the water pressure).
+        expected_cauchy_stress_yy = -1016.67
+
+        self.check_expected_outputs(
+            expected_stage_displacement_and_strain=-0.1,
+            expected_stress=expected_cauchy_stress_yy,
+            expected_total_displacement=-0.1,
+            output_data=output_data,
+            stage_nr=1,
```

Comment:
It seems that these two are always used together. Perhaps it's simpler to only pass the output data structure that is needed? That is, to merge these two into one parameter, where at the calling side the indexing is done:
```suggestion
            output_data=output_data[0],
```

A similar change is needed for the second, and we need to change method `check_expected_outputs`.

Reply:
Done, nice simplification!